test/test_charm.py

```diff
@@ -12,188 +12,155 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 import functools
-import os
 import pathlib
-import shutil
 import tempfile
 import typing
-import unittest
 from pathlib import Path
 
 import pytest
 import yaml
 
 import ops
 import ops.charm
-from ops.model import ModelError, _ModelBackend
-from ops.storage import SQLiteStorage
+from ops.model import ModelError
 
-from .test_helpers import fake_script, fake_script_calls
+from .test_helpers import FakeScript, create_framework
 
 
-class TestCharm(unittest.TestCase):
+@pytest.fixture
+def fake_script(request: pytest.FixtureRequest) -> FakeScript:
+    return FakeScript(request)
```

Comment:
We had discussed we were planning to import this from `test_helpers`, even if it meant a Pyright/lint ignore comment, to avoid the duplication.

Reply:
It's kept here temporarily, because there is a function with the same name in the `test_helpers.py`. When all files are refactored, it will be moved.